### ISSUE TITLE
Support the new Django 1.10 MIDDLEWARE style

### ISSUE THREAD

**Issue by tremby** [cobrateam/django-htmlmin#102]
I get a `TypeError` when using the django-htmlmin middlewares in `MIDDLEWARE` in Django 1.10 (they're fine when using the old-style `MIDDLEWARE_CLASSES`).

See https://docs.djangoproject.com/en/1.10/topics/http/middleware/#upgrading-pre-django-1-10-style-middleware for upgrading information.

**Comment by muminoff** [comment]
@tremby Did you find workaround for this?

**Comment by tremby** [comment]
Yes, just continue to use it the old way.

**Comment by bartmika** [comment]
I've just encountered this same problem.

+1

**Comment by bartmika** [comment]
I have found how to **solve** this problem. The new **middleware.py** should look like this and it would work:

```python
# Copyright 2013 django-htmlmin authors. All rights reserved.
# Use of this source code is governed by a BSD-style
# license that can be found in the LICENSE file.

import re

from django.conf import settings
from htmlmin.minify import html_minify


class MarkRequestMiddleware(object):
    def __init__(self, get_response):
        self.get_response = get_response
        # One-time configuration and initialization.

    def __call__(self, request):
        request._hit_htmlmin = True
        response = self.get_response(request)
        return response

class HtmlMinifyMiddleware(object):
    def __init__(self, get_response):
        self.get_response = get_response
        # One-time configuration and initialization.

    def __call__(self, request):
        response = self.get_response(request)

        minify = getattr(settings, "HTML_MINIFY", not settings.DEBUG)
        keep_comments = getattr(settings, 'KEEP_COMMENTS_ON_MINIFYING', False)
        parser = getattr(settings, 'HTML_MIN_PARSER', 'html5lib')
        if minify and self.can_minify_response(request, response):
            response.content = html_minify(response.content,
                                           ignore_comments=not keep_comments,
                                           parser=parser)
        return response

    def can_minify_response(self, request, response):
        try:
            req_ok = request._hit_htmlmin
        except AttributeError:
            return False

        if hasattr(settings, 'EXCLUDE_FROM_MINIFYING'):
            for url_pattern in settings.EXCLUDE_FROM_MINIFYING:
                regex = re.compile(url_pattern)
                if regex.match(request.path.lstrip('/')):
                    req_ok = False
                    break

        resp_ok = 'text/html' in response.get('Content-Type', '')
        if hasattr(response, 'minify_response'):
            resp_ok = resp_ok and response.minify_response
        return req_ok and resp_ok
```

**Comment by rodolfomartinez** [comment]
+1

**Comment by andrewsmedina** [comment]
fixed and released in 0.10.0 version

**Comment by muminoff** [comment]
👍
